copier/cli.py

```diff
@@ -103,7 +103,9 @@ class CopierApp(cli.Application):
 class _Subcommand(cli.Application):
     """Base class for Copier subcommands."""
 
-    data: AnyByStrDict = {}
+    def __init__(self, executable):
+        self.data: AnyByStrDict = {}
```

Comment:
Why this change?

Reply:
when `data` is a _class_ member, test data from previous test runs is unintentionally shared between all tests in the `test_cli.py` test suite.  I've pushed https://github.com/copier-org/copier/pull/1332/commits/ed656c5aabedd414dfa1458d2961c099dde2a21e to demo the CI failure that occurs.  

this failure won't show up if you run just run the failing test on its own (i.e. `pytest tests/test_cli.py -k 'test_data_file_parsed_by_question_type' `); it only shows up when tests before it are ran.

making `data` an _instance_ member solves this issue.  I'm guessing that this only matters for the test suites, since every time you invoke `copier` at the command line it spins up all state "fresh".

Reply:
Ah, mutability... I wish I had the time to rewrite copier in haskell 😂

Thanks!